distributions/otelcol-contrib/README.md

```diff
@@ -0,0 +1,16 @@
+# OpenTelemetry Collector Contrib Distro
+
+This distribution contains all the components from both the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector) repository and the [OpenTelemetry Collector Contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib) repository. This distribution includes open source and vendor supported components.
+
+## Recommendation
+
+As this distribution contains many components, it is a good starting point to try various configurations. However, when running in production, it is recommended to limit the collector to contain only the components necessary for an environment. Some reasons to do this:
+
+* reduce the size of the collector, reducing deployment times for the collector
+* improve the security of the collector by reducing the available attack surface area
+
+Building a custom collector can be achieved using the [OpenTelemetry Collector Builder](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder).
```

Comment:
Is the builder itself ready for production usage? It definitely does the job, since we use it for official releases, but 
- we don't have docs about it on https://opentelemetry.io
- we don't have 'ocb' Docker images
- people need to add `replace` statements to actually make the thing build, which we don't document anywhere AFAICT

I am not opposed in principle to this PR, but I think we need to improve our docs and packaging for this to be a feasible solution for many people, and also make sure that we are promoting usage of something that is mature enough

Reply:
I opened open-telemetry/opentelemetry-collector/issues/5713 and open-telemetry/opentelemetry-collector/issues/5712 to track a couple of these.

Reply:
> we don't have docs about it on https://opentelemetry.io/

There is this doc: https://opentelemetry.io/docs/collector/distributions/#custom-distributions

Reply:
> There is this doc: https://opentelemetry.io/docs/collector/distributions/#custom-distributions

There's actually this doc https://opentelemetry.io/docs/collector/custom-collector/ I was wrong about that one :) I think the other points still apply though

Reply:
> I think the other points still apply though

I agree

Reply:
I am good with moving this forward, but it also matches my opinions on contrib's distro haha

Reply:
I am good with moving forward too, I opened the issues a while back so this works for me :)